lib/Controller/AppointmentConfigController.php

```diff
@@ -22,72 +23,67 @@
  * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
  *
  */
+
 namespace OCA\Calendar\Controller;
 
 use OCA\Calendar\Db\AppointmentConfig;
+use OCA\Calendar\Exception\ClientException;
 use OCA\Calendar\Exception\ServiceException;
 use OCA\Calendar\Http\JsonResponse;
 use OCA\Calendar\Service\Appointments\AppointmentConfigService;
 use OCP\AppFramework\Controller;
-use OCP\AppFramework\Http\TemplateResponse;
-use OCP\IInitialStateService;
+use OCP\AppFramework\Http;
 use OCP\IRequest;
-use OCP\IUser;
 
-/**
- * Class PublicViewController
- *
- * @package OCA\Calendar\Controller
- */
 class AppointmentConfigController extends Controller {
 
-	/** @var IInitialStateService */
-	private $initialStateService;
-
-	/** @var IUser */
-	private $user;
-
 	/** @var AppointmentConfigService */
 	private $appointmentConfigService;
 
-	/**
-	 * @param string $appName
-	 * @param IRequest $request an instance of the request
-	 * @param IInitialStateService $initialStateService
-	 * @param IUser $user
-	 * @param AppointmentConfigService $appointmentService
-	 */
-	public function __construct(string                   $appName,
-								IRequest                 $request,
-								IInitialStateService     $initialStateService,
-								IUser                    $user,
-								AppointmentConfigService $appointmentService) {
+	/** @var string|null */
+	private $userId;
+
+	public function __construct(string $appName,
+								IRequest $request,
+								AppointmentConfigService $appointmentService,
+								?string $userId) {
```

Comment:
Or we can forbid null for $userId here.

Reply:
doesn't work, then accessing the route anonymously triggers a type error. we first instantiate the controller before method permissions (`@PublicPage`) is checked